### PR TITLE
Tezos: fix operations on originated accounts

### PR DESCRIPTION
**Pull request by KhalilBellakrid**
This a patch to allow actions from originated account.
We rely on RPC call not to make serialization more complex than it is and also because those accounts are considered as legacy.
This PR was tested with Live and it's working : https://tzstats.com/KT1PTC8w6Mk6MSnx56TzMRx4Z8weLLq3oGgP

### CODE REVIEW

File: core/src/wallet/common/api_impl/OperationApi.cpp [hunk 1]
```diff
@@ -70,7 +68,7 @@ namespace ledger {
             return _backend.recipients;
         }
 
-        Operation &OperationApi::getBackend() {
+        ledger::core::Operation &OperationApi::getBackend() {
```
Comment (phaazon) [inline core/src/wallet/common/api_impl/OperationApi.cpp]: We’re in the same namespace; why is that needed?

File: core/src/wallet/tezos/TezosLikeAccount2.cpp [hunk 2]
```diff
@@ -206,11 +207,10 @@ namespace ledger {
                                           return seq.str();
                                       })
                     .recoverWith(getContext(), [] (const Exception &e) {
+                        // Avoid fmt error because of JSON string
+                        auto error = "Failed to broadcast raw transaction, reason " + e.getMessage();
```
Comment (alekece) [inline core/src/wallet/tezos/TezosLikeAccount2.cpp]: Don't forget `const` qualifier

File: core/src/wallet/tezos/TezosLikeAccount2.cpp [hunk 3]
```diff
@@ -206,11 +207,10 @@ namespace ledger {
...
                         return Future<std::string>::failure(
-                                make_exception(
-                                        api::ErrorCode::INCOMPLETE_TRANSACTION,
-                                        fmt::format("Failed to broadcast raw transaction, reason {}", e.getMessage())
-                                )
+                                Exception(api::ErrorCode::INCOMPLETE_TRANSACTION, error)
```
Comment (alekece) [inline core/src/wallet/tezos/TezosLikeAccount2.cpp]: What is the advantage to use `make_exception` over `Exception` ?

File: core/src/wallet/tezos/TezosLikeAccount2.cpp [hunk 4]
```diff
@@ -243,22 +244,31 @@ namespace ledger {
                 tx->setSender(accountAddress);
                 tx->setReceiver(TezosLikeAddress::fromBase58(request.toAddress, currency));
                 tx->setSigningPubKey(self->getKeychain()->getPublicKey().getValue());
+                tx->setManagerAddress(managerAddress);
                 tx->setType(request.type);
-                return explorer->getCounter(senderAddress).flatMapPtr<Block>(self->getContext(), [self, tx, explorer] (const std::shared_ptr<BigInt> &counter) {
+                auto counterAddress = protocolUpdate == api::TezosConfigurationDefaults::TEZOS_PROTOCOL_UPDATE_BABYLON ?
```
Comment (alekece) [inline core/src/wallet/tezos/TezosLikeAccount2.cpp]: Ditto

File: core/src/wallet/tezos/api_impl/TezosLikeTransactionApi.h [hunk 5]
```diff
@@ -103,6 +102,10 @@ namespace ledger {
 
             TezosLikeTransactionApi & setBalance(const BigInt &balance);
 
+            TezosLikeTransactionApi & setManagerAddress(const std::string &managerAddress);
+            std::string getManagerAddress();
```
Comment (alekece) [inline core/src/wallet/tezos/api_impl/TezosLikeTransactionApi.h]: Ditto

File: core/src/wallet/tezos/TezosLikeAccount2.cpp (hunk 3 above)
Comment (KhalilBellakrid) [reply core/src/wallet/tezos/TezosLikeAccount2.cpp]: Not going through `fmt` which is crashing if you don't escape braces ! :trollface:

Comment (alekece) [reply core/src/wallet/tezos/TezosLikeAccount2.cpp]: You cannot use here `make_exception(api::ErrorCode::INCOMPLETE_TRANSACTION, error)` ?

Comment (KhalilBellakrid) [reply core/src/wallet/tezos/TezosLikeAccount2.cpp]: Please read my comment above (in the code too) and `make_exception` code ! 👌

File: core/src/wallet/common/api_impl/OperationApi.cpp (hunk 1 above)
Comment (KhalilBellakrid) [reply core/src/wallet/common/api_impl/OperationApi.cpp]: It was just a 'clean up' to comply with declaration ! (and we can't get rid of the one in declaration because of some namespaces clash 😞)
